app/src/main/java/com/github/sdpcoachme/profile/CoachesListActivity.kt

```diff
@@ -11,45 +11,69 @@ import androidx.compose.foundation.layout.*
 import androidx.compose.foundation.shape.CircleShape
-import androidx.compose.material.Divider
-import androidx.compose.material.Icon
-import androidx.compose.material.MaterialTheme
-import androidx.compose.material.Text
-import androidx.compose.material.icons.Icons
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons.Default
 import androidx.compose.material.icons.filled.Place
+import androidx.compose.material.icons.filled.Tune
 import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.unit.dp
 import com.github.sdpcoachme.CoachMeApplication
 import com.github.sdpcoachme.R
+import com.github.sdpcoachme.data.Sports
 import com.github.sdpcoachme.data.UserInfo
 import com.github.sdpcoachme.errorhandling.ErrorHandlerLauncher
 import com.github.sdpcoachme.location.provider.FusedLocationProvider.Companion.CAMPUS
 import com.github.sdpcoachme.messaging.ChatActivity
+import com.github.sdpcoachme.profile.CoachesListActivity.TestTags.Buttons.Companion.FILTER
 import com.github.sdpcoachme.ui.Dashboard
 import com.github.sdpcoachme.ui.theme.CoachMeTheme
+import com.github.sdpcoachme.ui.theme.Purple200
 import kotlinx.coroutines.future.await
+import java.util.*
 import java.util.concurrent.CompletableFuture
 
 class CoachesListActivity : ComponentActivity() {
+
+    class TestTags {
+        class Buttons {
+            companion object {
+                const val FILTER = "filterSearch"
+            }
+        }
+    }
+
     // Allows to notice testing framework that the activity is ready
     var stateLoading = CompletableFuture<Void>()
+
+    // Observable state of the current sports used to filter the coaches list
+    private lateinit var sportsFilter: MutableState<List<Sports>>
+    private lateinit var selectSportsHandler: (Intent) -> CompletableFuture<List<Sports>>
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         val isViewingContacts = intent.getBooleanExtra("isViewingContacts", false)
         val database = (application as CoachMeApplication).database
-        val lastLocation = (application as CoachMeApplication).locationProvider.getLastLocation()
-        val userLatLng = lastLocation.value?: CAMPUS
         val email = database.getCurrentEmail()
 
+        val locationProvider = (application as CoachMeApplication).locationProvider
+        // Here we don't need the UserInfo
+        locationProvider.init(this, CompletableFuture.completedFuture(null))
+        val userLatLng = locationProvider.getLastLocation().value?: CAMPUS
```

Comment:
Adding a small comment explaining when getLastLocation can return null could be helpful to better understand this small bit of code.